test/thermo/PengRobinson_Test.cpp

```diff
@@ -0,0 +1,316 @@
+        standardConcs[0] = test_phase->standardConcentration(0);
+        standardConcs[2] = test_phase->standardConcentration(2);
+        test_phase->getActivityConcentrations(&activityConcentrations[0]);
+
+        EXPECT_NEAR(standardConcs[0] * r * activityCoeffs[0], activityConcentrations[0], 1.e-6);
+        EXPECT_NEAR(standardConcs[2] * (1-r) * activityCoeffs[2], activityConcentrations[2], 1.e-6);
+    }
+}
+
+TEST_F(PengRobinson_Test, setTP)
+{
+    // Check to make sure that the phase diagram is accurately reproduced for a few select isobars
+
+    // All sub-cooled liquid:
+    const double p1[6] = {
+        6.6603507723749249e+002,
+        1.6824762614489907e+002,
+        1.6248354709581241e+002,
+        1.5746729362032696e+002,
+        1.5302217175386241e+002,
+        1.4902908974486667e+002
+    };
+    // Phase change between temperatures 4 & 5:
+    const double p2[6] = {
+        7.5732259810273172e+002,
+        7.2766981078381912e+002,
+        6.935475475396446e+002,
+        6.5227027102964917e+002,
+        5.9657442842753153e+002,
+        3.9966973266966875e+002
+    };
+    // Supercritical; no discontinuity in rho values:
+    const double p3[6] = {
+        8.0601205067780199e+002,
+        7.8427655940884574e+002,
+        7.6105347579146576e+002,
+        7.3605202492828505e+002,
+        7.0887891410210011e+002,
+        6.7898591969734434e+002
+    };
+
+    for(int i=0; i<6; ++i)
+    {
+        const double temp = 294 + i*2;
+        set_r(0.999);
+        test_phase->setState_TP(temp, 5542027.5);
+        EXPECT_NEAR(test_phase->density(),p1[i],1.e-8);
+
+        test_phase->setState_TP(temp, 7388370.);
+        EXPECT_NEAR(test_phase->density(),p2[i],1.e-8);
+
+        test_phase->setState_TP(temp, 9236712.5);
+        EXPECT_NEAR(test_phase->density(),p3[i],1.e-8);
+    }
+}
+
+TEST_F(PengRobinson_Test, getPressure)
+{
+    // Check to make sure that the P-R equation is accurately reproduced for a few selected values
+
+    /* This test uses CO2 as the only species (mole fraction 100%).
+    *  Values of a_coeff, b_coeff are calculated based on the the critical temperature
+    *  and pressure values of CO2 as follows:
+    *       a_coeff = 0.457235(RT_crit)^2/p_crit
+    *       b_coeff = 0.077796(RT_crit)/p_crit
+    *  The temperature dependent parameter in P-R EoS is calculated as
+    *       \alpha = [1 + \kappa(1 - sqrt{T/T_crit}]^2
+    *  kappa is a function calulated based on the accentric factor.
+    */
+
+    double a_coeff = 3.958095109E+5;
+    double b_coeff = 26.62616317/1000;
+    double acc_factor = 0.228;
+    double pres_theoretical, kappa, alpha, mv;
+    const double rho = 1.0737; 
+
+    //Calculate kappa value 
+    kappa = 0.37464 + 1.54226*acc_factor - 0.26992*acc_factor*acc_factor;
+
+    for (int i = 0; i<10; i++)
+    {
+        const double temp = 296 + i * 2;
+        set_r(1.0);        
+        test_phase->setState_TR(temp, rho);
```

Comment:
I like this test, but it could cover a lot wider range of T and rho, couldn't it?

Reply:
Modified the test to cover a wider range.